### PR TITLE
[#4708] improvement(client-java/server): Add implementations for the `getFileLocation` interface in Java Client / Server

### PR DESCRIPTION
**Pull request by xloya**
### What changes were proposed in this pull request?

Add the implementations for `getFileLocation` interfaces in Java Client / Server.

### Why are the changes needed?

Fix: #4708 

### How was this patch tested?

Add some UTs and ITs.

### CODE REVIEW

**Comment by xloya** [comment]
@jerryshao Hi, this PR is ready for review, please take a look when you have time, thanks.

File: clients/client-java/src/main/java/org/apache/gravitino/client/FilesetCatalog.java [hunk 1]
```diff
@@ -238,7 +241,22 @@ public boolean dropFileset(NameIdentifier ident) {
   @Override
   public String getFileLocation(NameIdentifier ident, String subPath)
       throws NoSuchFilesetException {
-    throw new UnsupportedOperationException("Not implemented");
+    checkFilesetNameIdentifier(ident);
+    Namespace fullNamespace = getFilesetFullNamespace(ident.namespace());
+
+    CallerContext callerContext = CallerContext.CallerContextHolder.get();
+    Map<String, String> queryParams = Maps.newHashMap();
+    queryParams.put("subPath", subPath);
```
Comment (jerryshao) [inline clients/client-java/src/main/java/org/apache/gravitino/client/FilesetCatalog.java:249]: I think maybe we should encoding the subPath to avoid supported chars in the URL.

File: clients/client-java/src/main/java/org/apache/gravitino/client/FilesetCatalog.java [hunk 2]
```diff
@@ -238,7 +241,22 @@ public boolean dropFileset(NameIdentifier ident) {
...
+    FileLocationResponse resp =
+        restClient.get(
+            formatFilesetRequestPath(fullNamespace) + "/" + ident.name() + "/" + "fileLocation",
```
Comment (jerryshao) [inline clients/client-java/src/main/java/org/apache/gravitino/client/FilesetCatalog.java:252]: Is it better to add the th subPath into the URL, not a queryParam?

File: core/src/main/java/org/apache/gravitino/catalog/FilesetNormalizeDispatcher.java [hunk 3]
```diff
@@ -96,7 +96,9 @@ public boolean dropFileset(NameIdentifier ident) {
 
   @Override
   public String getFileLocation(NameIdentifier ident, String subPath) {
-    throw new UnsupportedOperationException("Not implemented");
+    // The constraints of the name spec may be more strict than underlying catalog,
+    // and for compatibility reasons, we only apply case-sensitive capabilities here.
+    return dispatcher.getFileLocation(normalizeCaseSensitive(ident), subPath);
```
Comment (jerryshao) [inline core/src/main/java/org/apache/gravitino/catalog/FilesetNormalizeDispatcher.java:101]: I would suggest split the core part logic into another PR. In this PR, we only focus on client and server side API. Don't make the PR too big, which is hard to review.

File: clients/client-java/src/main/java/org/apache/gravitino/client/FilesetCatalog.java (hunk 1 above)
Comment (xloya) [reply clients/client-java/src/main/java/org/apache/gravitino/client/FilesetCatalog.java]: Done

File: clients/client-java/src/main/java/org/apache/gravitino/client/FilesetCatalog.java (hunk 2 above)
Comment (xloya) [reply clients/client-java/src/main/java/org/apache/gravitino/client/FilesetCatalog.java]: Hi, I prefer to use the `queryParams` for the following reasons:
1. Currently other similar GET APIs pass parameters in the form of queryParams, like `RelationalTable.listPartitions()`;
2. I looked at the APIs of other commercial products, such as the REST API of Databricks' DBFS, and they also pass these parameters through queryParams: https://docs.databricks.com/api/workspace/dbfs/read;
3. In the unit test, I tried to directly splice the URL, but it seems that the test framework cannot mock the request, so I guess some frameworks may not recognize the method of directly splicing parameters to the URL.

File: core/src/main/java/org/apache/gravitino/catalog/FilesetNormalizeDispatcher.java (hunk 3 above)
Comment (xloya) [reply core/src/main/java/org/apache/gravitino/catalog/FilesetNormalizeDispatcher.java]: Done, this PR has been split to only include changes with the Java Client and Server.

File: clients/client-java/src/main/java/org/apache/gravitino/client/FilesetCatalog.java [hunk 4]
```diff
@@ -238,7 +241,23 @@ public boolean dropFileset(NameIdentifier ident) {
...
   public String getFileLocation(NameIdentifier ident, String subPath)
       throws NoSuchFilesetException {
-    throw new UnsupportedOperationException("Not implemented");
+    checkFilesetNameIdentifier(ident);
+    Namespace fullNamespace = getFilesetFullNamespace(ident.namespace());
+
+    CallerContext callerContext = CallerContext.CallerContextHolder.get();
+
+    Map<String, String> params = new HashMap<>();
+    params.put("subPath", RESTUtils.encodeString(subPath));
```
Comment (jerryshao) [inline clients/client-java/src/main/java/org/apache/gravitino/client/FilesetCatalog.java:250]: I would suggest we don't use camel style for param here "subPath" and path "fileLocation", how about "sub_path" and "location"?

Comment (xloya) [reply clients/client-java/src/main/java/org/apache/gravitino/client/FilesetCatalog.java]: Fixed

**Review by jerryshao — APPROVED** [review]
